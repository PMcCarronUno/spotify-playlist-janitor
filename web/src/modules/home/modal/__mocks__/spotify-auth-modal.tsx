import React from 'react';
export const SpotifyAuthModalView = () => <div>Spotify Auth Modal View</div>
