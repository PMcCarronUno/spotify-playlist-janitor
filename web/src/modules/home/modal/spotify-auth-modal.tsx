import styled from "styled-components";
import { SubTitle, Text } from "shared/components/typography";

export const SpotifyAuthModalView = () => {
  return (
    <ModalContainer>
      <SubTitle style={{ marginBottom: 8 }}>
        Spotify Authentication Error
      </SubTitle>
      <Text>
        The application has not been authenticated with your Spotify account.
        Please follow{" "}
        <a
          href={process.env.REACT_APP_API_URL}
          target="_blank"
          rel="noreferrer"
        >
          this
        </a>{" "}
        link and sign into Spotify, and <a href=".">reload</a> this page.
      </Text>
    </ModalContainer>
  );
};

const ModalContainer = styled.div`
  width: 360px;
`;

SpotifyAuthModalView.displayName = "SpotifyAuthModalView";
