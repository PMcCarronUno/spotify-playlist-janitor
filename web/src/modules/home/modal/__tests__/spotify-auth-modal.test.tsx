import { fireEvent, getByTestId, render } from "@testing-library/react";

import { SpotifyAuthModalView } from "../spotify-auth-modal";

describe("<SpotifyAuthModalView />", () => {
  let container: HTMLElement;

  beforeEach(async () => {
    ({ container } = render(<SpotifyAuthModalView />));
  });

  beforeAll(() => {
    process.env.REACT_APP_API_URL = 'https://localhost:5001';
  });

  it("should render SpotifyAuthModalView component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
