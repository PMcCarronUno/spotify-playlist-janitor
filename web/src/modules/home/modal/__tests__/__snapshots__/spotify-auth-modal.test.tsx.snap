// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpotifyAuthModalView /> should render SpotifyAuthModalView component 1`] = `
<div
  class="sc-jTQDJr wuNdd"
>
  <div
    class="sc-guDMob jLqGJD"
    style="margin-bottom: 8px;"
  >
    Spotify Authentication Error
  </div>
  <div
    class="sc-dmyDmy dcXriK"
  >
    The application has not been authenticated with your Spotify account. Please follow
     
    <a
      href="https://localhost:5001"
      rel="noreferrer"
      target="_blank"
    >
      this
    </a>
     
    link and sign into Spotify, and 
    <a
      href="."
    >
      reload
    </a>
     this page.
  </div>
</div>
`;
